Sources/Parsers/ColorsFileParser.swift

```diff
@@ -4,34 +4,70 @@
 // MIT Licence
 //
 
-import AppKit.NSColor
 import Foundation
-import Kanna
 import PathKit
 
-public protocol ColorsFileParser {
-  var colors: [String: UInt32] { get }
-}
-
 public enum ColorsParserError: Error, CustomStringConvertible {
   case invalidHexColor(string: String, key: String?)
   case invalidFile(reason: String)
+  case unsupportedFileType(supported: [String])
 
   public var description: String {
     switch self {
-    case .invalidHexColor(string: let string, key: let key):
+    case .invalidHexColor(let string, let key):
       let keyInfo = key.flatMap { " for key \"\($0)\"" } ?? ""
       return "error: Invalid hex color \"\(string)\" found\(keyInfo)."
-    case .invalidFile(reason: let reason):
+    case .invalidFile(let reason):
       return "error: Unable to parse file. \(reason)"
+    case .unsupportedFileType(let supported):
+      return "error: Unsupported file type. The supported file types are: \(supported.joined(separator: ", "))"
+    }
+  }
+}
+
+protocol ColorsFileTypeParser: class {
+  static var extensions: [String] { get }
+
+  init()
+  func parseFile(at path: Path) throws -> [String: UInt32]
+}
+
+public final class ColorsFileParser {
+  private var parsers = [String: ColorsFileTypeParser.Type]()
+  var colors = [String: UInt32]()
+
+  public init() {
+    register(parser: ColorsCLRFileParser.self)
+    register(parser: ColorsJSONFileParser.self)
+    register(parser: ColorsTextFileParser.self)
+    register(parser: ColorsXMLFileParser.self)
+  }
+
+  public func parseFile(at path: Path) throws {
+    guard let parserType = parsers[path.extension?.lowercased() ?? ""] else {
+      throw ColorsParserError.unsupportedFileType(supported: Array(parsers.keys))
+    }
+
+    let parser = parserType.init()
+    let colors = try parser.parseFile(at: path)
+
+    for (name, value) in colors {
+      self.colors[name] = value
```

Comment:
🤔 yeah indeed probably not SwiftGenKit's job. 

SwiftGenKit should only either throw (in case of fatal error) or fill an array of warning messages that consumers of the library could then inspect. Or call a callback/closure that the consumer can provide, like a delegate pattern.